blog/_posts/2023-04-19-conda-r-geospatial.md

```diff
@@ -0,0 +1,87 @@
+Conda provides a [number of commands for outputting a specification file](https://arctraining.github.io/hpc2-software/course/conda.html#sharing-conda-environments) for your environment that you can shared and have others reproduce your environment from. 
+We **strongly** recommend that if you're using Conda you create these environment specification files and include them in your project repositories to help reproduce your environment. 
+However that is a topic outside of the scope of this particular blog post so lets now get back to installing `terra`.
+
+With our environment created we now need to enter that environment with our shell. 
+Conda environments are these isolated bubbles which we need to step into and out of in the shell in order for that isolation to work.
+To enter a Conda environment and have access to all the packages we've installed in it we use the command `conda activate ENV_NAME`, where `ENV_NAME` is the name of the environment. 
+However, on ARC4 we use the older version of this command `source activate ENV_NAME` to avoid some niggles we have with the newer behaviour of `conda activate`.
+
+```bash
+# activate the environment to use the packages you've installed
+source activate r-geo
+```
+
+After running `source activate` our prompt will be prepended with the name of our environment and we can now access the packages we installed into this environment.
+To install our R package we can either start up the R console and run `install.packages` directly or use the `R -e` command to pass in a string to the R to be run without opening a full R console.
+
+```bash
+# Install packages in R without starting an R console
+(r-geo) R -e 'install.packages("terra");' 
+```
+
+This will now attempt to install the package using the version of R we installed with Conda. 
+This ensures that all the libraries we installed through Conda will be available to R as it attempts to build any R packages that have C++ dependencies and uses the same compiler toolchain that was used to compile the version of R we installed with Conda.
+Any packages installed within a Conda environment are also saved within the Conda environment, helping isolate these R packages from any other R packages you may have installed using an R module or in another Conda environment.
+This again really helps with isolating our environment to prevent any package conflicts that occur and also makes it easy to dispose of our environment when we're done with it (we just delete the conda environment!).
+
+Overall, Conda is a very useful tool for managing dependencies and as demonstrated here is not just for Python!
+We can use it to manage versions of R, along with library dependencies for R packages. 
+This however does come with some potential drawbacks:
+- We've added another tool, by using Conda to manage dependencies we've added to the complexity of our project because for people to reproduce it they will now need Conda
```

Comment:
```suggestion
- We've added another tool; by using Conda to manage dependencies we've added to the complexity of our project because for people to reproduce it they will now need Conda.
```